serverless.yml

```diff
@@ -20,5 +50,121 @@ functions:
         - lib/**
         - node_modules/**
     events:
-      - http: ANY /
-      - http: ANY /{proxy+}
+      - http:
+          path: /
+          method: ANY
+          authorizer: ${self:custom.authorizer.${self:service}-authorizer}
+      - http:
+          path: /{proxy+}
+          method: ANY
+          authorizer: ${self:custom.authorizer.${self:service}-authorizer}
+
+  hn-evidence-store-authorizer:
+    name: ${self:service}-${self:provider.stage}-authorizer
+    handler: authorizer.handler
+    package:
+      include:
+        - authorizer.js
+        - node_modules/**
+    environment:
+      ALLOWED_GROUPS: 'housingneeds-singleview-beta'
+      JWT_SECRET: ${ssm:/common/hackney-jwt-secret}
+
+resources:
+  Resources:
+    StoreBucket:
+      Type: AWS::S3::Bucket
+      Properties:
+        BucketName: ${self:provider.environment.STORE_BUCKET}
+        PublicAccessBlockConfiguration:
+          BlockPublicAcls: true
+          BlockPublicPolicy: true
+          IgnorePublicAcls: true
+          RestrictPublicBuckets: true
+        BucketEncryption:
+          ServerSideEncryptionConfiguration:
+            - ServerSideEncryptionByDefault:
+                SSEAlgorithm: AES256
+
+    ElasticSearchInstance:
+      Type: AWS::Elasticsearch::Domain
+      Properties:
+        EBSOptions:
+          EBSEnabled: true
+          VolumeType: gp2
+          VolumeSize: 10
+        ElasticsearchClusterConfig:
+          InstanceType: t2.small.elasticsearch
+          InstanceCount: 1
+          DedicatedMasterEnabled: false
+          ZoneAwarenessEnabled: false
+        ElasticsearchVersion: 7.4
+        AccessPolicies:
+          Version: '2012-10-17'
+          Statement:
+            Action:
+              - es:ESHttpGet
+              - es:ESHttpPut
+              - es:ESHttpPost
+              - es:ESHttpHead
+            Principal: '*'
+            Effect: Allow
+
+    # CloudFrontDistribution:
+    #   Type: AWS::CloudFront::Distribution
+    #   Properties:
+    #     DistributionConfig:
+    #       Aliases:
+    #         - ${self:custom.aliases.${self:provider.stage}}
+    #       PriceClass: PriceClass_100
+    #       ViewerCertificate:
```

Comment:
API Gateway supports custom domains, not sure if we really need a CloudFront distribution?